gadfly/sentence_summarizer.py

```diff
@@ -58,3 +59,92 @@ def summarize(self, sents, n):
+            all_tokens = [token.lemma_ for sent in sents for token in sent]
+        else:
+            print("Please enter an appropriate option for 'style'.")
+        document_freq_dict = Counter(all_tokens)
+        return document_freq_dict, all_tokens
+
+    def get_dict_tf_idf(self, sents, document_freq_dict, all_tokens, style):
+        """Get a dict of _____"""
+        dict_tf_idf = {}
+        for word in set(all_tokens):
+            for n, sent in enumerate(sents):
+                sum = 0
+                if style == 'lower':
+                    if word in [token.lower_ for token in sent]:
+                        sum += 1
+                elif style == 'lemma':
+                    if word in [token.lemma_ for token in sent]:
+                        sum += 1
+                else:
+                    if word in [token.text for token in sent]:
+                        sum += 1
+                dict_tf_idf[(n, word)] = sum * math.log(len(sents)/document_freq_dict[word])
+        return dict_tf_idf
+
+    def get_ranked_sents(self, lensen, dict_tf_idf):
+        n_scores = [0] * lensen
+        for (n, word), value in dict_tf_idf.items():
+            n_scores[n] = n_scores[n] + value
+        ranked_sents = (sorted([(x, n) for n, x in enumerate(n_scores)]))
+        return ranked_sents
+
+    def get_text(self, sents):
+        sents_text = []
+        for span in sents:
+            tokens = []
+            for token in span:
+                tokens.append(token.text)
+            sents_text.append(tokens)
+        return sents_text
+
+    def EDA(self, sents_text, dict_tf_idf):
+        eda_data = defaultdict(list)
+        for (index, token), value in dict_tf_idf.items():
+            if value > 0:
+                eda_data[index].append((round(value,2), token))
+        EDA_sents_text = []
+        for n, text in enumerate(sents_text):
+            EDA_sents_text.append(text)
+            EDA_sents_text[n] = EDA_sents_text[n] + ['\nEDA:\n'] + [str(sorted(eda_data[n], key=lambda x: x[0], reverse=True))]
+        return EDA_sents_text
+
+    def summarize(self, sents, n):
+        assert n <= len(sents)
+        style = 'lemma'
+        document_freq_dict, all_tokens = self.get_tokens_and_freqs(sents, 
+                                                                   style)
+        dict_tf_idf = self.get_dict_tf_idf(sents, 
+                                           document_freq_dict, 
+                                           all_tokens, 
+                                           style)
+        ranked_sents = self.get_ranked_sents(len(sents), 
+                                             dict_tf_idf)
+        sents_text = self.get_text(sents)
+        EDA = 0
+        if EDA:
+            sents_text = self.EDA(sents_text, dict_tf_idf)
+
+        top_sentences = []
+        for score, index in ranked_sents[-1:-(n+1):-1]:
+            top_sentences.append(sents_text[index])
+
+        for i in range(n):

```

Comment:
Yeah. All of these are distinct tokens that should be removed as early as possible I think.